### PR TITLE
fix kw hits regarding to overflow issue

### PR DESCRIPTION
**Pull request by YarShev**

### CODE REVIEW

**Comment by SmirnovEgorRu** [comment]
make test

**Comment by YarShev** [comment]
make test

**Comment by YarShev** [comment]
> please fix clear code issue.
> Precommit is failed.

Fixed

File: service/kernel/error_handling.cpp [hunk 1]
```diff
@@ -105,19 +105,21 @@ class MessageCollection : public Collection<SharedPtr<Message<IDType> > >
         bool found = false;
         size_t index = 0;
 
-        if(this->size() == 0) { return find(_noMessageFound); }
+        if (this->size() == 0)
+        {
+            return SharedPtr<Message<IDType>>(new Message<IDType>(_noMessageFound, "No error message found"));
```
Comment (ShvetsKS) [inline service/kernel/error_handling.cpp]: NoErrorMessageFound should be used here.

File: service/kernel/error_handling.cpp [hunk 2]
```diff
@@ -105,19 +105,21 @@ class MessageCollection : public Collection<SharedPtr<Message<IDType> > >
...
+            if ((*this)[i]->id() == id)
             {
                 found = true;
                 index = i;
             }
         }
 
-        if(found) { return (*this)[index]; }
-        else { return find(_noMessageFound); }
+        return found ? (*this)[index] : SharedPtr<Message<IDType>>(new Message<IDType>(_noMessageFound, "No error message found"));
```
Comment (ShvetsKS) [inline service/kernel/error_handling.cpp]: the same as above

File: service/kernel/error_handling.cpp (hunk 1 above)
Comment (YarShev) [reply service/kernel/error_handling.cpp]: Fixed

File: service/kernel/error_handling.cpp (hunk 2 above)
Comment (YarShev) [reply service/kernel/error_handling.cpp]: Fixed

**Comment by YarShev** [comment]
make test

**Comment by YarShev** [comment]
make test

**Comment by YarShev** [comment]
make test

**Comment by YarShev** [comment]
make test

**Pull request merged**